File: traffic-gen/scenarios/test/phased-http-dns.lua
-- python3 traffic-gen/snowtg.py <this-file> -- <EAL> -- <app-options>
local tg = require("snowtg")
local peer = os.getenv("SNOWTG_PEER") or "192.168.10.234"

return tg.scenario("gateway-phased-http-dns", {concurrency=256,
    phases={
        tg.phase("warmup",    3, 200),
        tg.phase("ramp_up",   3, 1000, {start=200}),
        tg.phase("steady",    4, 1000),
        tg.phase("spike",     2, 2000),
        tg.phase("cool_down", 3, 200),
    },
    classes={
        tg.http("http_keepalive", peer, 8888, {weight=6, keepalive=true}),
        tg.http("http_short", peer, 8888, {weight=2}),
        tg.dns("dns", peer, "snowtg.test", 1053, {weight=2}),
    }})

File: traffic-gen/scenarios/test/phased-http-dns.py
"""Run via: python3 traffic-gen/snowtg.py <this-file> -- <EAL> -- <app-options>."""
import os
from snowtg import scenario, phase, http, dns

peer = os.environ.get("SNOWTG_PEER", "192.168.10.234")
plan = scenario("gateway-phased-http-dns", concurrency=256,
    phases=[
        phase("warmup",    3, 200),
        phase("ramp_up",   3, 1000, start=200),
        phase("steady",    4, 1000),
        phase("spike",     2, 2000),
        phase("cool_down", 3, 200),
    ],
    classes=[
        http("http_keepalive", peer, 8888, weight=6, keepalive=True),
        http("http_short", peer, 8888, weight=2),
        dns("dns", peer, "snowtg.test", 1053, weight=2),
    ])
